library/junos_ping

```diff
@@ -0,0 +1,248 @@
+        HAS_PYEZ = False
+    else:
+        HAS_PYEZ = True
+except ImportError:
+    HAS_PYEZ = False
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            host=dict(required=True),
+            logfile=dict(required=False, default=None),
+            user=dict(required=False, default=os.getenv('USER')),
+            passwd=dict(required=False, default=None),
+            port=dict(required=False, default=830),
+            dest_ip=dict(required=True, default=None),
+            source_ip=dict(required=False, default=None),
+            interface=dict(required=False, default=None),
+            rapid=dict(required=False, default=True),
+            routing_instance=dict(required=False, default=None),
+            ttl=dict(required=False, default=None),
+            count=dict(required=False, default='5')
+            ),
+        supports_check_mode=False
+    )
+
+    if not HAS_PYEZ:
+        module.fail_json(msg='junos-eznc >= 1.2.2 is required for this module')
+
+    results = {}
+    m_args = module.params
+
+    # Open connection to device
+    dev = Device(
+            m_args['host'],
+            user=m_args['user'],
+            passwd=m_args['passwd'],
+            port=m_args['port'],
+            gather_facts=False)
+
+    try:
+        dev.open()
+    except Exception as err:
+        msg = 'Unable to connect to {0}: {1}'.format(m_args['host'], str(err))
+        module.fail_json(msg=msg)
+        return
+
+    results['dest_ip'] = m_args['dest_ip']
+    results['count'] = m_args['count']
+    results['rapid'] = m_args['rapid']
+
+    results['changed'] = False
+
+    # Prepare parameters
+    ping_params = dict(
+            host=m_args['dest_ip'],
+            count=m_args['count'],
+            rapid=m_args['rapid'])
+
+    if m_args['source_ip'] is not None:
+        ping_params['source'] = m_args['source_ip']
+        results['source_ip'] = m_args['source_ip']
+
+    if m_args['routing_instance'] is not None:
+        ping_params['routing_instance'] = m_args['routing_instance']
+        results['routing_instance'] = m_args['routing_instance']
+
+    if m_args['interface'] is not None:
+        ping_params['interface'] = m_args['interface']
+        results['interface'] = m_args['interface']
+
+    if m_args['ttl'] is not None:
+        ping_params['ttl'] = m_args['ttl']
+        results['ttl'] = m_args['ttl']
+
+    try:
+
+        # Execute Ping
+        rpc_reply = dev.rpc.ping(**ping_params)
+
+        # results['xml_reply'] = etree.tostring(rpc_reply)
+
+        # Try to Find probe summary
+        probe_summary = rpc_reply.find('probe-results-summary')
+        if probe_summary is None:
+            results['failed'] = True
+
+            error_message = rpc_reply.find('rpc-error/error-message').text
+            if error_message is not None:
+                msg = error_message.strip()
+
+            logging.error(msg)
+            dev.close()
+            module.fail_json(msg=msg)
```

Comment:
If error_message is None, this will raise an exception because msg has not been assigned.